### ISSUE TITLE
Does Turbolinks have to check location's extension before visiting it?

### ISSUE THREAD

**Issue by akxcv** [turbolinks/turbolinks#26]
Hello! I've got the following issue with Turbolinks: if I try to visit a URL like `/chris.martin`, Turbolinks doesn't perform a Visit and the whole page gets reloaded. If I remove the dot from this URL (`/chrismartin`), Turbolinks performs a Visit as it should.

In the Turbolinks source code:
```coffeescript
# controller.coffee
locationIsVisitable: (location) ->
  location.isPrefixedBy(@getRootLocation()) and location.isHTML()

# location.coffee
getExtension: ->
  @getLastPathComponent().match(/\.[^.]*$/)?[0]

isHTML: ->
  extension = @getExtension()
  extension is ".html" or not extension?
```

So my question is: Why does Turbolinks check URL's extension before visiting it? Maybe we need to add an optional attribute like `data-turbolinks-extension="false"`? Or is there a better way do solve this problem?

**EDIT**: Now that I think about it, this extension check makes sense (what if I request a raw .js file, for example?). But I think we need an option to disable it on specific links.

**Comment by akxcv** [comment]
If `data-turbolinks-extension="false"` is a good solution to this problem, I'm ready to create a pull request.

**Comment by akxcv** [comment]
#29

**Comment by akxcv** [comment]
#29

**Comment by packagethief** [comment]
Hey @akxcv, we discussed this last week in #15.

tldr; we think applications should be able to define what constitutes an "HTML URL." Until there's official API for that, you can [override Turbolinks.Location.prototype.isHTML](https://github.com/turbolinks/turbolinks/issues/15#issuecomment-186639159).
